app/src/screens/SignIn/index.tsx

```diff
@@ -61,10 +64,33 @@ const styles = StyleSheet.create({
     height: 20,
     marginLeft: 4,
   },
+  errorWarning: {
+    height: 20,
+    width: 20,
+  },
+  errorContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    width: '90%',
+  },
+  errorText: {
+    paddingLeft: 4,
+    lineHeight: 20,
+    fontSize: 14,
+    color: '#D61F1F',
+  },
 });
 
 const SignIn: React.FC = () => {
   const dispatch = useDispatch<Dispatch>();
+  const errorMessage = useSelector((state: State) => {
+    if (state.voice.accessToken.status === 'rejected') {
+      switch (state.voice.accessToken.reason) {
+        case 'TOKEN_RESPONSE_NOT_OK':
```

Comment:
Is this the only failure-state we need to handle?

Reply:
There should probably be a `default` fall-through here or we should be more exhaustive with cases for each reason. Whatever is simpler and still satisfies the compiler. Perhaps some boolean value out of the state on whether or not we should show the error at all too.

Reply:
yup only fail-state that we need to handle

Reply:
What happens for `FETCH_ERROR`, `FETCH_TEXT_ERROR`, or `USER_NOT_FULFILLED`? Do we want these errors to effectively be transparent for the end user?

I also think that we should still collapse the branches on line 63 and 71 into one. What about something like
```
if (!tokenResponse.ok) {
  const error = tokenTextResult.status === 'fulfilled' ? new Error(tokenTextResult.value) : tokenTextResult.reason;
  return rejectWithValue({
    ...,
    error: miniSerializeError(error),
  });
}
```
That way we don't have multiple branches repeating some similar stuff. This should work too because if `.text()` fulfills, then it will return a string, which we can put into an error object for serialization. And if it rejects, then the reason is already an error object that can be serialized.

Reply:
Oops, left the comment in the wrong file. It's lines 63 and 71 in the `store/voice/accessToken` file.

Reply:
done